dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -136,75 +135,40 @@ def get_columns_for_unique_arrays(self, time_dim, load_data_df):
-        expected_timestamps = time_range.list_time_range()
-        expected_count = len(expected_timestamps)
-
-        groupby_cols = self.get_columns_for_unique_arrays(time_dim, load_data_df)
-        time_col = time_dim.get_timestamp_load_data_columns()
-        timestamps_by_dims = (
-            load_data_df[time_col + groupby_cols]
-            .groupby(groupby_cols)
-            .agg(F.countDistinct(*time_col).alias("distinct_timestamps"))
-        )
-
-        distinct_counts = timestamps_by_dims.select("distinct_timestamps").distinct()
-        if distinct_counts.count() > 1:
-            for row in timestamps_by_dims.collect():
-                if row.distinct_timestamps != len(expected_timestamps):
-                    logger.error(
-                        "load_data row with dimensions=%s does not have %s %s: actual=%s",
-                        self._show_selected_keys_from_dict(row, groupby_cols),
-                        len(expected_timestamps),
-                        time_col[0],
-                        row.distinct_timestamps,
-                    )
-
-                    raise DSGInvalidDataset(
-                        f"One or more arrays do not have {len(expected_timestamps)} timestamps"
-                    )
-        else:
-            val = distinct_counts.collect()[0].distinct_timestamps
-            if val != expected_count:
-                raise DSGInvalidDataset(
-                    f"load_data arrays do not have {len(expected_timestamps)} {time_col[0]}: actual={val}"
-                )
-
-    @staticmethod
-    def _show_selected_keys_from_dict(dct, keys: list):
-        """Return a subset of a dictionary based on a list of keys.
-
-        Returns
-        -------
-        Dict
-
-        """
-        dct_selected = []
-        for key in keys:
-            dct_selected.append(dct[key])
-        return dict(zip(keys, dct_selected))
+            self._check_dataset_time_consistency_by_time_array(time_dim, load_data_df)
+
+    @track_timing(timer_stats_collector)
+    def _check_dataset_time_consistency_by_time_array(self, time_dim, load_data_df):
+        """Check that each unique time array has the same timestamps."""
+        logger.info("Check dataset time consistency by time array.")
+        unique_array_cols = self.get_columns_for_unique_arrays(time_dim, load_data_df)
+        time_cols = time_dim.get_timestamp_load_data_columns()
+        counts = load_data_df.groupBy(*time_cols).count().select("count")
+        distinct_counts = counts.select("count").distinct().collect()
+        if len(distinct_counts) != 1:
+            raise DSGInvalidDataset(
+                f"All time arrays must have the same times: unique timestamp counts = {len(distinct_counts)}"
+            )
+        unique_ta_counts = load_data_df.select(*unique_array_cols).distinct().count()
+        if unique_ta_counts != distinct_counts[0]["count"]:
+            raise DSGInvalidDataset("dataset has invalid counts")
```

Comment:
NIT, but more specific msg is better for debugging. sth like this?
f"dataset expects {unique_ta_counts} count for each timestamp in {time_col[0]}: actual={distinct_counts[0]["count"]}"